src/main.c

```diff
@@ -814,7 +817,32 @@ int main(int argc, char *argv[]) {
 #endif
 
   // Process command line arguments
-  while((ch = getopt(argc, argv, "?Ab:B:c:C:DeE:Fi:l:nNp:OP:qrtT:U:vVx:")) != -1) {
+  struct option longopts[] = {
+    {"help",       no_argument,       NULL, '?'},
+    {"baud",       required_argument, NULL, 'b'},
+    {"bitclock",   required_argument, NULL, 'B'},
+    {"programmer", required_argument, NULL, 'c'},
+    {"config",     required_argument, NULL, 'C'},
+    {"noerase",    no_argument,       NULL, 'D'},
+    {"erase",      no_argument,       NULL, 'e'},
+    {"logfile",    required_argument, NULL, 'l'},
+    {"test",       no_argument,       NULL, 'n'},
+    {"noconfig",   no_argument,       NULL, 'N'},
+    {"part",       required_argument, NULL, 'p'},
+    {"chip",       required_argument, NULL, 'p'},
+    {"port",       required_argument, NULL, 'P'},
+    {"quiet",      no_argument,       NULL, 'q'},
+    {"reconnect",  no_argument,       NULL, 'r'},
+    {"terminal",   no_argument,       NULL, 't'},
+    {"tty",        no_argument,       NULL, 't'},
```

Comment:
Now that I look at the list carefully, I don't think `tty` is a good long option name to signify `-t` for the principal reason that tty describes a communication method (teletype), which is orthogonal to what `-t` does. Terminal input can come from a file, from a pipe, via a USB connection etc and not necessarily from a tty. I'd prefer to either have no second long option for `-t`, but *if* there is an appetite for a second option, I'd much rather have one of `--interactive`, `--console`, `--avr-console`, `--shell` or similar. We normally use the wording terminal or interactive terminal in the documentation for `-t`, so `--terminal` might be sufficient.

Could we add `--terminal-command` for `-T` (note capital T), which like `-U` is not interactive?

Reply:
"shell" implies turning completeness. "interactive" captures the spirit well.